RELEASE.md

```diff
@@ -47,13 +48,17 @@ See the next section for details on cutting an individual release.
 The quick and easy way to create a GitHub issue using the following template and follow the instructions from the issue itself.
 If something is not clear, you can get back to this document to learn more about the process.
 
+> **Note:** Branches with `release-` prefix in the name are protected, use some different branch name for making your PRs, for example `my/release-notes`.
+
 ````markdown
 ### Publish the release candidate
 
 - [ ] Begin drafting the [release notes](https://github.com/grafana/mimir/blob/main/RELEASE.md#write-release-notes-document)
-  - Write release notes to `main`, then cherry pick them into the release branch
-  - Don't block on it to publish the release candidate
+  - Create the release note PR targeting the main branch
+  - This step shouldn't block from publishing release candidate
+  - After the release note PR is merged (which usually happen after RC is published), cherry-pick them into the release branch
```

Comment:
```suggestion
  - Create the release notes PR targeting the main branch
  - This step shouldn't block from publishing release candidate
  - After the release notes PR is merged (which usually happen after RC is published), cherry-pick them into the release branch
```

I think we refer to the whole document as release notes. 

Also maybe an example PR may be useful 

```suggestion
  - Create the release notes PR targeting the main branch; [example PR](https://github.com/grafana/mimir/pull/4740)
  - This step shouldn't block from publishing release candidate
  - After the release notes PR is merged (which usually happen after RC is published), cherry-pick them into the release branch
```